### PR TITLE
Add API docs for /operations and /operations{id} endpoints.

### PR DESCRIPTION
**Pull request by CDJellen**
## Description

Add API documentation for the `/operations` and `/operations/{id}` endpoints.

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

This PR includes summaries and descriptions for the `/operations` and `/operations/{id}` endpoints in the Swagger UI.  The branch was tested by navigating to the Swagger documentation page and verifying that all summaries and descriptions rendered correctly, with proper formatting.

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: app/api/v2/handlers/operation_api.py [hunk 1]
```diff
@@ -34,39 +34,53 @@ def add_routes(self, app: web.Application):
...
-    @aiohttp_apispec.docs(tags=['operations'])
-    @aiohttp_apispec.request_schema(OperationSchema(partial=True, only=['state',
-                                                                        'autonomous',
-                                                                        'obfuscator']))
+    @aiohttp_apispec.docs(tags=['operations'],
+                          summary='Update fields within an operation',
+                          description='Update one CALDERA operation in memory based on the operation id (String '
+                                      'UUID). Any fields in the operation object may be edited in the request body '
+                                      'using the `OperationSchema`.')
+    @aiohttp_apispec.request_schema(OperationSchema(partial=True))
```
Comment (bleepbop) [inline app/api/v2/handlers/operation_api.py:74]: This is the only endpoint where this specific change needs to be made, but can you add back the `only=['state', 'autonomous', 'obfuscator']` to the `request_schema`? The `store()` method of `c_operation.py` is called whenever we issue a PATCH/update request, and the only fields that can be updated in that method are state, autonomous, and obfuscator

File: app/api/v2/handlers/operation_api.py [hunk 2]
```diff
@@ -34,39 +34,53 @@ def add_routes(self, app: web.Application):
...
     async def get_operations(self, request: web.Request):
         operations = await self.get_all_objects(request)
         return web.json_response(operations)
 
-    @aiohttp_apispec.docs(tags=['operations'])
+    @aiohttp_apispec.docs(tags=['operations'],
+                          summary='Retrieve an operation by operation id',
+                          description='Retrieve one CALDERA operation from memory based on the operation id (String '
+                                      'UUID).  Use fields from the `BaseGetOneQuerySchema` in the request body to add '
+                                      '`include` and `exclude` filters.')
```
Comment (bleepbop) [inline app/api/v2/handlers/operation_api.py:51]: For the endpoints that contain a parameter in the url (GET operation by ID, PATCH, PUT, and DELETE) you can add the following to the `aiohttp_apispec.docs` decorator to describe the url parameter:
```
parameters=[{
  'in': 'path',
  'name': 'id',
  'schema': {'type': 'string'},
  'required': 'true',
  'description': 'UUID of the Operation object to be retrieved.'
}],
```

File: app/api/v2/handlers/operation_api.py [hunk 3]
```diff
@@ -34,39 +34,53 @@ def add_routes(self, app: web.Application):
...
     async def update_operation(self, request: web.Request):
         operation = await self.update_object(request)
         return web.json_response(operation.display)
 
-    @aiohttp_apispec.docs(tags=['operations'])
+    @aiohttp_apispec.docs(tags=['operations'],
+                          summary='Delete an operation by operation id',
+                          description='Delete one CALDERA operation from memory based on the operation id (String '
+                                      'UUID).')
     @aiohttp_apispec.response_schema(OperationSchema)
```
Comment (bleepbop) [inline app/api/v2/handlers/operation_api.py:84]: One last thing that might be nice to have is a description for the `response_schema` in every endpoint. While GET operation by ID, PUT, PATCH, and POST all return a single dumped Operation object, `GET /operations` and `DELETE /operations{id}` return a list of dumped objects and a 204 HTTP response code, respectively.

**Review by bleepbop — CHANGES_REQUESTED** [review]

**Review by elegantmoose — APPROVED** [review]
LGTM as long as @bleepbop comments are addressed.

**Review by elegantmoose — DISMISSED** [review]

File: app/api/v2/handlers/operation_api.py [hunk 4]
```diff
@@ -34,40 +34,80 @@ def add_routes(self, app: web.Application):
...
+                          parameters=[{
+                              'in': 'path',
+                              'name': 'id',
+                              'schema': {'type': 'string'},
+                              'required': 'true',
+                              'description': 'UUID of the Operation object to be retrieved.'
+                          }])
     @aiohttp_apispec.querystring_schema(BaseGetOneQuerySchema)
-    @aiohttp_apispec.response_schema(OperationSchema(partial=True))
+    @aiohttp_apispec.response_schema(OperationSchema(partial=True, only=['state', 'autonomous', 'obfuscator']),
```
Comment (bleepbop) [inline app/api/v2/handlers/operation_api.py:61]: I think the `only=['state', 'autonomous', 'obfuscator']` was accidentally added to this endpoint - can you remove that from this endpoint, and add that to the `response_schema` for `update_operation()`?

File: app/api/v2/handlers/operation_api.py [hunk 5]
```diff
@@ -122,7 +162,7 @@ async def get_operation_link_result(self, request: web.Request):
         return web.json_response(result)
 
     @aiohttp_apispec.docs(tags=['operations'])
-    @aiohttp_apispec.request_schema(LinkSchema(partial=True, only=['command', 'status']))
+    @aiohttp_apispec.request_schema(LinkSchema(partial=True))
```
Comment (bleepbop) [inline app/api/v2/handlers/operation_api.py:165]: I don't think that these are part of your ticket, but could you add back `only=['command', 'status']` just for this endpoint?

**Review by bleepbop — CHANGES_REQUESTED** [review]
Just a few tiny changes and it should be good to go!

**Comment by CDJellen** [comment]
Thank you for your review Ben, I've made the changes in the latest commit.

File: app/api/v2/handlers/operation_api.py [hunk 6]
```diff
@@ -34,40 +34,80 @@ def add_routes(self, app: web.Application):
...
+                                      'using the `OperationSchema`.',
+                          parameters=[{
+                              'in': 'path',
+                              'name': 'id',
+                              'schema': {'type': 'string'},
+                              'required': 'true',
+                              'description': 'UUID of the Operation object to be retrieved.'
+                          }])
+    @aiohttp_apispec.request_schema(OperationSchema(partial=True))
+    @aiohttp_apispec.response_schema(OperationSchema(partial=True, only=['state', 'autonomous', 'obfuscator']),
```
Comment (bleepbop) [inline app/api/v2/handlers/operation_api.py:92]: Sorry one last tiny change - can you move `only=['state', 'autonomous', 'obfuscator']` into the `request_schema` for this endpoint instead? I think this was accidentally added to the `response_schema`

File: app/api/v2/handlers/operation_api.py [hunk 7]
```diff
@@ -132,9 +172,7 @@ async def update_operation_link(self, request: web.Request):
         link = await self._api_manager.update_operation_link(operation_id, link_id, data, access)
         return web.json_response(link)
 
-    @aiohttp_apispec.docs(tags=['operations'], summary='The only required fields for this endpoint are "paw", '
-                                                       '"executor.name", "executor.command", and "executor.platform". '
-                                                       '"executor.command" is expected to be unencoded.')
```
Comment (bleepbop) [inline app/api/v2/handlers/operation_api.py:137]: Can you add this summary back for this endpoint?

**Review by bleepbop — APPROVED** [review]

**Review by elegantmoose — APPROVED** [review]

**Comment by elegantmoose** [comment]
@wbooth - I think this PR is gtg if you want to override the lagging CI.

**Review by wbooth — APPROVED** [review]

**Pull request merged**